### PR TITLE
Adds dot env support

### PR DESCRIPTION
**Pull request by paganotoni**
This PR adds `.env` support for generated buffalo apps, when generating a new buffalo app this adds an empty `.env` file, adds that file to the `.gitignore` and uses `envy.Load()` to load the `.env` every time the application restarts.

This PR also adds .env to the list of extensions that will be listened by refresh to restart the app, this is because it is cool to have the env change applied without stopping buffalo dev.

This is related with #606

### CODE REVIEW

File: generators/newapp/templates/actions/app.go.tmpl [hunk 1]
```diff
@@ -34,6 +34,9 @@ var T *i18n.Translator
 // should be defined. This is the nerve center of your
 // application.
 func App() *buffalo.App {
+  //Loading .env file values
+  envy.Load()
```
Comment (markbates) [inline generators/newapp/templates/actions/app.go.tmpl]: Move this into `buffalo.New` that way it'll just work for every app, and they don't need the extra line. Check to make sure that moving this doesn't cause any problems for older apps, as well as apps that don't have a `.env`.

**Comment by paganotoni** [comment]
@markbates thanks for the feedback, I changed the `eny.Load()` to be done in the buffalo.New and added a short description in the generated .env

**Comment by paganotoni** [comment]
@markbates what do you think about generating the SECRET_KEY on the .env by default ?

**Comment by markbates** [comment]
I don't think it's good idea. In development that key doesn't really matter, it only matters in production. From my understanding of the .env file is that it's not supposed to be committed, but is for each developer to set up for their environment. If that is correct we should also add it to the default .gitignore

**Comment by paganotoni** [comment]
Yes, it is ignored currently, right, was thinking developers to have environment as closer to production as possible, but you're right if it doesn't make a difference then I'll forget it 👍

**Comment by paganotoni** [comment]
Also, let me know if there is something else to do on this one and I'll jump into it, thanks for all the feedback!

**Pull request merged**